lib/portage/dbapi/bintree.py

```
@@ -199,25 +211,41 @@ def aux_update(self, cpv, values):
 				cpv = self._instance_key(cpv, support_string=True)[0]
 				build_id = cpv.build_id
 
-		tbz2path = self.bintree.getname(cpv)
-		if not os.path.exists(tbz2path):
+		binpkg_path = self.bintree.getname(cpv)
+		if not os.path.exists(binpkg_path):
 			raise KeyError(cpv)
-		mytbz2 = portage.xpak.tbz2(tbz2path)
-		mydata = mytbz2.get_data()
+
+		if binpkg_path.endswith(SUPPORTED_XPAK_EXTENSIONS):
+			mytbz2 = portage.xpak.tbz2(binpkg_path)
+			mydata = mytbz2.get_data()
+			encoding_key = True
+		elif binpkg_path.endswith(SUPPORTED_GPKG_EXTENSIONS):
+			mybinpkg = portage.gpkg.gpkg(self.settings, cpv, binpkg_path)
+			mydata = mybinpkg.get_metadata()
+			encoding_key = False
+		else:
+			return
 
 		for k, v in values.items():
-			k = _unicode_encode(k,
-				encoding=_encodings['repo.content'], errors='backslashreplace')
+			if encoding_key:
+				k = _unicode_encode(k,
+					encoding=_encodings['repo.content'], errors='backslashreplace')
 			v = _unicode_encode(v,
 				encoding=_encodings['repo.content'], errors='backslashreplace')
 			mydata[k] = v
 
 		for k, v in list(mydata.items()):
 			if not v:
 				del mydata[k]
-		mytbz2.recompose_mem(portage.xpak.xpak_mem(mydata))
+		if binpkg_path.endswith(SUPPORTED_XPAK_EXTENSIONS):
+			mytbz2.recompose_mem(portage.xpak.xpak_mem(mydata))
+		elif binpkg_path.endswith(SUPPORTED_GPKG_EXTENSIONS):
+			mybinpkg.update_metadata(mydata)
+		else:
+			return
```

We should raise InvalidBinaryPackageFormat here.